### PR TITLE
Changed sign up order

### PR DESCRIPTION
**Pull request by lerabo**
-Added role and changed sign up order(firstly sign up, choose role and after post request)

### CODE REVIEW

File: src/pages/RoleSelection.tsx [hunk 1]
```diff
@@ -2,16 +2,21 @@ import { FC } from 'react';
...
 
 const RoleSelection: FC = () => {
 	const { t } = useTranslation();
 	const navigate = useNavigate();
-	const dispatch = useDispatch();
+	const dispatch = useAppDispatch();
+	const [signUp] = useSignUpMutation();
 	const role = useSelector((state: RootState) => state.user.role);
+	const email = useSelector((state: RootState) => state.user.email);
+	const password = useSelector((state: RootState) => state.user.password);
```
Comment (lyudad) [inline src/pages/RoleSelection.tsx]: why we select 3 values role, email, pass if we just can get user? it make component bigger and I don't really see reason

File: src/pages/RoleSelection.tsx [hunk 2]
```diff
@@ -21,8 +26,16 @@ const RoleSelection: FC = () => {
 	const handleChange = (event: RadioChangeEvent) => {
 		dispatch(saveRole(event.target.value));
 	};
-	const handleClick = async () => {
-		navigate('/sign-up');
+	const handleClick = async (role: string) => {
+		try {
+			const res = await signUp({ email, password, role: role }).unwrap();
+			localStorage.setItem('userId', JSON.stringify(res.id));
```
Comment (lyudad) [inline src/pages/RoleSelection.tsx]: can we use redux-persist instead?

File: src/pages/RoleSelection.tsx (hunk 1 above)
Comment (lerabo) [reply src/pages/RoleSelection.tsx]: removed

File: src/pages/RoleSelection.tsx (hunk 2 above)
Comment (lerabo) [reply src/pages/RoleSelection.tsx]: used

**Pull request closed without merging**